semperfiwebdesign/all-in-one-seo-pack
# Fix preview snippet bugs

Issue #1408 & #2526.

## Proposed changes

Fixes two bugs for the preview snippet title.

(1) Double % signs in the title break the title (#2526)
(2) Double site title on Posts page (#1408)

The former was happening because there was some code that was deleting everything in between two percentage signs (including the signs); the latter was happening because the title for the Posts page is already handled by get_aioseop_title().

In the meanwhile, I've also cleaned up the phpDocs and refactored a little bit.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- Improves existing code

## Checklist


## Testing instructions

For #2526:
Add two % signs to the title of a new Post with some text in between each % sign and make sure the preview snippet displays the title correctly.

For #1408:
Set a static Posts page under Settings > Reading, navigate to it and make sure the preview snippet doesn't include the site title twice.
